cmd/ax/upgrade.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"archive/tar"
+	"bufio"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"github.com/kardianos/osext"
+)
+
+func getLatestReleaseData() *github.RepositoryRelease {
+	client := github.NewClient(nil)
+
+	release, _, err := client.Repositories.GetLatestRelease(context.Background(), "egnyte", "ax")
+	if err != nil {
+		fmt.Printf("Couldn't get a release information")
+		os.Exit(1)
+	}
+
+	return release
+}
+
+func getLatestAssetLink() string {
+	release := getLatestReleaseData()
+	var assetDownloadLink string
+
+	for _, element := range release.Assets {
+		assetLink := element.GetBrowserDownloadURL()
+		if strings.Contains(assetLink, runtime.GOOS) && strings.Contains(assetLink, runtime.GOARCH) {
+			assetDownloadLink = assetLink
+			break
+		}
+	}
+
+	return assetDownloadLink
+}
+
+func getLatestReleaseTag() string {
+	release := getLatestReleaseData()
+	return release.GetTagName()
+}
+
+func downloadFile(filepath string, url string) error {
+
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func upgradeVersion() error {
+	if latestVersion := getLatestReleaseTag(); version != latestVersion {
```

Comment:
I'd probably do this to avoid a giant indented block:
```
latestVersion := getLatestReleaseTag()
if version == latestVersion {
   fmt.Println("Latest version is already installed.")
   return
}
// Upgrade case here
```

Reply:
I was considering it as I would the same in Python :joy: Im going to improve that.